lib/Basics/files-advanced.cpp

```diff
@@ -0,0 +1,320 @@
+////////////////////////////////////////////////////////////////////////////////
```

Comment:
We could split this file into two, (well, even 3 or more :/), but what I caught on was that one bit of the file is under `USE_ENTERPRISE` and the other is  not; if that is split we can do away with the ifdef, maybe even move the code into the `enterprise` repo.


(I am happy for that not to necessarily happen in this PR!)